README.md

```diff
@@ -13,19 +72,114 @@ class ProjectRecord < ApplicationRecord
   self.table_name = "projects"
 
   include MuchSlug::ActiveRecord
-  has_slug({
-    :source       => proc{ "#{self.id}-#{self.abbrev}" },
-    :preprocessor => :upcase
-  })
-  before_update :reset_slug, :if => :abbrev_changed?
+  has_slug(
+    source: -> { "#{self.id}-#{self.name}" },
+  )
+  has_slug(
+    attribute: :full_slug
+    source:    -> { "#{self.id}-#{self.full_name}" },
+  )
 
   # ...
 end
 ```
 
-## Sequel
+#### Preprocessor
+
+By default, MuchSlug doesn't pre-process the slug value source before generating the slug value. You can specify a custom pre-processor by passing any Proc-like object:
 
-TODO
+```ruby
+require "much-slug/activerecord"
+
+class ProjectRecord < ApplicationRecord
+  self.table_name = "projects"
+
+  include MuchSlug::ActiveRecord
+  has_slug(
+    source:       -> { "#{self.id}-#{self.name}" },
+    preprocessor: :downcase
+  )
+  has_slug(
+    attribute:    :full_slug
+    source:       -> { "#{self.id}-#{self.full_name}" },
+    preprocessor: -> { |source_value| source_value[0..30] }
+  )
+
+  # ...
+end
+```
+
+#### Separator
+
+MuchSlug replaces any non-word characters with a separator. This helps make slugs URL-friendly. By default, MuchSlug uses `"-"` for the separator. You can specify a custom separator value when configuring slugs:
+
+```ruby
+require "much-slug/activerecord"
+
+class ProjectRecord < ApplicationRecord
+  self.table_name = "projects"
+
+  include MuchSlug::ActiveRecord
+  has_slug(
+    source:    -> { "#{self.id}.#{self.name}" },
+    separator: "."
+  )
+
+  # ...
+end
+
+project = ProjectRecord.last
+project.id   # => 123
+project.name # => "Sprockets 2.0"
+project.save
+project.slug # => "123.Sprockets.2.0"
+```
+
+#### Allowing Underscores
+
+By default, MuchSlug doesn't allow underscores in source values and treats them like non-word characters. This means it replaces underscores with the separator. You can override this to allow underscores when configuring slugs
```

Comment:
@kellyredding - I think you've changed this default so that it does allow underscores?

Reply:
Sorry for the confusion - I changed back to not allowing underscores by default. It is one of the commits in this. I decided to switch back b/c it didn't feel great as I was writing usage docs in the README.

Reply:
My bad, I read that in your PR description and then totally forgot about it when reading the README.